### ISSUE TITLE
Double quoting false positive for `$COMP_CWORD`

### ISSUE THREAD

**Issue by hyperupcall** [koalaman/shellcheck#2732]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
# shellcheck shell=bash

echo $COMP_CWORD
```

#### Here's what shellcheck currently says:

```sh
[Line 3:](javascript:setPosition(3, 6))
echo $COMP_CWORD
     ^-- [SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting.
```

#### Here's what I wanted or expected to see:

Nothing

#### Extra Details

The Bash manpage states this:

> COMP_CWORD
> An  index  into ${COMP_WORDS} of the word containing the current cursor position.  This variable is available only in shell functions  invoked  by  the  programmable completion facilities (see Programmable Completion below).

Since the value is always an integer (like $?), I think this error is a false positive
